src/win/winsock.c

```diff
@@ -131,21 +131,14 @@ void uv_winsock_init(void) {
   if (dummy != INVALID_SOCKET) {
     opt_len = (int) sizeof protocol_info;
     if (getsockopt(dummy,
-                   SOL_SOCKET,
-                   SO_PROTOCOL_INFOW,
-                   (char*) &protocol_info,
-                   &opt_len) == SOCKET_ERROR)
-      uv_fatal_error(WSAGetLastError(), "getsockopt");
-
-    if (!(protocol_info.dwServiceFlags1 & XP1_IFS_HANDLES))
-      uv_tcp_non_ifs_lsp_ipv6 = 1;
-
-    if (closesocket(dummy) == SOCKET_ERROR)
-      uv_fatal_error(WSAGetLastError(), "closesocket");
-
-  } else if (!error_means_no_support(WSAGetLastError())) {
-    /* Any error other than "socket type not supported" is fatal. */
-    uv_fatal_error(WSAGetLastError(), "socket");
+        SOL_SOCKET,
+        SO_PROTOCOL_INFOW,
+        (char*)&protocol_info,
+        &opt_len) == 0) {
+        if (!(protocol_info.dwServiceFlags1 & XP1_IFS_HANDLES))
+            uv_tcp_non_ifs_lsp_ipv6 = 1;
+    }
+    closesocket(dummy);
```

Comment:
Shouldn't the closesocket() call still check for SOCKET_ERROR?

Reply:
I think the return value is useless for a client application, we could do nothing when OS ipv6 configuration is not set correctly sometimes. Maybe the most import thing is Don't let it Crash :)

Reply:
That's not the libuv philosophy. Unexpected errors shouldn't go unnoticed.

Reply:
IPv6 socket is optional, we can use IPv4 instead,  so check of invalid IPv6 socket shouldn't be a fatal error.
It's not Unexpected errors.